code/game/atoms_movable.dm

```diff
@@ -649,3 +649,205 @@
 		.["Remove deadchat control"] = "?_src_=vars;removedeadchatcontrol=[UID()]"
 
+/atom/movable/proc/choose_crush_crit(mob/living/carbon/victim)
+	if(!length(GLOB.tilt_crits))
+		return
+	for(var/crit_path in shuffle(GLOB.tilt_crits))
+		var/datum/tilt_crit/C = GLOB.tilt_crits[crit_path]
+		if(C.is_valid(src, victim))
+			return C
+
+
+/atom/movable/proc/handle_squish_carbon(mob/living/carbon/victim, damage_to_deal, datum/tilt_crit/crit)
+
+	// Damage points to "refund", if a crit already beats the shit out of you we can shelve some of the extra damage.
+	var/crit_rebate = 0
+
+	if(HAS_TRAIT(victim, TRAIT_DWARF))
+		// also double damage if you're short
+		damage_to_deal *= 2
+
+	if(crit)
+		crit_rebate = crit.tip_crit_effect(src, victim)
+		if(crit.harmless)
+			return
+
+		add_attack_logs(null, victim, "critically crushed by [src] causing [crit]")
+	else
+		add_attack_logs(null, victim, "crushed by [src]")
+
+	// 30% chance to spread damage across the entire body, 70% chance to target two limbs in particular
+	damage_to_deal = max(damage_to_deal - crit_rebate, 0)
+	if(prob(30))
+		victim.apply_damage(damage_to_deal, BRUTE, BODY_ZONE_CHEST, spread_damage = TRUE)
+	else
+		var/picked_zone
+		var/num_parts_to_pick = 2
+		for(var/i = 1 to num_parts_to_pick)
+			picked_zone = pick(BODY_ZONE_CHEST, BODY_ZONE_HEAD, BODY_ZONE_L_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_ARM, BODY_ZONE_R_LEG)
+			victim.apply_damage((damage_to_deal) * (1 / num_parts_to_pick), BRUTE, picked_zone)
+
+	victim.AddElement(/datum/element/squish, 80 SECONDS)
+
+#define NO_CRUSH_DIR "no_dir"
+
+/**
+ * Tip over this atom onto a turf, crushing things in its path.
+ *
+ * Arguments:
+ * * target_turf - The turf to fall onto.
+ * * should_crit - If true, we'll try to crit things that we crush.
+ * * crit_damage_factor - If a crit is rolled, crush_damage will be multiplied by this amount.
+ * * forced_crit - If passed, this crit will be applied to everything it crushes.
+ * * weaken_time - The amount of time that weaken will be applied to crushed mobs.
+ * * knockdown_time - The amount of time that knockdown will be applied to crushed mobs.
+ * * ignore_gravity - If false, we won't fall over in zero G.
+ * * should_rotate - If false, we won't rotate when we fall.
+ * * angle - The angle by which we'll rotate. If this is null/0, we'll randomly rotate 90 degrees clockwise or counterclockwise.
+ * * rightable - If true, the tilted component will be applied, allowing people to alt-click to right it.
+ * * block_interactions_until_righted - If true, interactions with the object will be blocked until it's righted.
+ * * crush_dir - The direction we're crushing.
+ */
+/atom/movable/proc/fall_and_crush(turf/target_turf, crush_damage, should_crit = FALSE, crit_damage_factor = 2, datum/tilt_crit/forced_crit, weaken_time = 4 SECONDS, knockdown_time = 10 SECONDS, ignore_gravity = FALSE, should_rotate = TRUE, angle, rightable = FALSE, block_interactions_until_righted = FALSE, crush_dir = NO_CRUSH_DIR)
```

Comment:
![image](https://github.com/ParadiseSS13/Paradise/assets/77684085/a05ab533-f1b7-4fd2-ac8e-899ec8854b9b)

Reply:
Yeah, this is pretty rough, but I'm not sure of a better way to do this (except maybe a tip_config datum)